source/solver_class.cc

```diff
@@ -53,27 +65,145 @@ flag_matrices_were_updated(true)
 
 
 
+template<int dim>
+void SolverBase<dim>::clear()
+{
+  flag_matrices_were_updated = true;
+}
+
+
+
+
 template<int dim>
 ProjectionSolverBase<dim>::ProjectionSolverBase(
-  TimeDiscretization::VSIMEXMethod                &time_stepping,
-  const std::shared_ptr<Mapping<dim>>             external_mapping,
-  const std::shared_ptr<ConditionalOStream>       external_pcout,
-  const std::shared_ptr<TimerOutput>              external_timer)
+  const RunTimeParameters::ProjectionSolverParametersBase &parameters,
+  TimeDiscretization::VSIMEXMethod                        &time_stepping,
+  std::shared_ptr<Entities::FE_VectorField<dim>>          &vector_field,
+  std::shared_ptr<Entities::FE_ScalarField<dim>>          &lagrange_multiplier,
+  const std::shared_ptr<Mapping<dim>>                     external_mapping,
+  const std::shared_ptr<ConditionalOStream>               external_pcout,
+  const std::shared_ptr<TimerOutput>                      external_timer)
 :
 SolverBase<dim>(time_stepping,
                 external_mapping,
                 external_pcout,
                 external_timer),
+projection_solver_parameters(parameters),
+vector_field(vector_field),
+lagrange_multiplier(lagrange_multiplier),
 norm_diffusion_step_rhs(std::numeric_limits<double>::lowest()),
 norm_projection_step_rhs(std::numeric_limits<double>::lowest()),
-flag_setup_auxiliary_scalar(true)
+flag_setup_auxiliary_scalar(true),
+flag_mean_value_constrain(false)
+{
+  // Explicitly set the supply term's pointer to null
+  ptr_supply_term = nullptr;
+}
+
+
+
+template<int dim>
+ProjectionSolverBase<dim>::ProjectionSolverBase(
+  const RunTimeParameters::ProjectionSolverParametersBase &parameters,
+  TimeDiscretization::VSIMEXMethod                        &time_stepping,
+  std::shared_ptr<Entities::FE_VectorField<dim>>          &vector_field,
+  const std::shared_ptr<Mapping<dim>>                     external_mapping,
+  const std::shared_ptr<ConditionalOStream>               external_pcout,
+  const std::shared_ptr<TimerOutput>                      external_timer)
+:
+SolverBase<dim>(time_stepping,
+                external_mapping,
+                external_pcout,
+                external_timer),
+projection_solver_parameters(parameters),
+vector_field(vector_field),
+norm_diffusion_step_rhs(std::numeric_limits<double>::lowest()),
+norm_projection_step_rhs(std::numeric_limits<double>::lowest()),
+flag_setup_auxiliary_scalar(true),
+flag_mean_value_constrain(false)
 {
   // Explicitly set the supply term's pointer to null
   ptr_supply_term = nullptr;
 }
```

Comment:
Why don't you call the default constructor for the `ptr_supply_term`?

Reply:
`ptr_supply_term` is a raw pointer. I am not aware of raw pointers having a default constructor. They either initialized or not initialized, if I am not mistaken.